NoClassDefFoundError: org/junit/runner/JUnitCore

I can run tests in a single module project, but as soon as I have multiple modules (e.g. opening up https://github.com/spring-cloud/spring-cloud-function/tree/master/spring-cloud-function-samples since that was what I was working on), I get this:

```
Error occured: Exception in thread "main" Error occured: java.lang.NoClassDefFoundError: org/junit/runner/JUnitCore
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
Error occured: 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at com.microsoft.java.test.runner.JUnitLauncher.main(JUnitLauncher.java:20)
Caused by: java.lang.ClassNotFoundException: org.junit.runner.JUnitCore
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 13 more
```

--- a/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/ProjectUtils.java
+++ b/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/internal/ProjectUtils.java
@@ -11,6 +11,7 @@
 package com.microsoft.java.test.plugin.internal;
 
 import java.net.URI;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.Set;
 import java.util.stream.Collectors;
@@ -20,6 +21,8 @@ import org.eclipse.core.resources.IProject;
 import org.eclipse.core.resources.IWorkspaceRoot;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
 import org.eclipse.jdt.core.IJavaProject;
 import org.eclipse.jdt.core.JavaCore;
 
@@ -27,14 +30,13 @@ public class ProjectUtils {
 	
 	public static Set<IJavaProject> parseProjects(URI rootFolderURI) {
 		IWorkspaceRoot workspaceRoot = ResourcesPlugin.getWorkspace().getRoot();
-		IContainer[] containers = workspaceRoot.findContainersForLocationURI(rootFolderURI);
-		return Arrays.stream(containers).map(container -> {
-			if (container instanceof IProject) {
-				return getJavaProject((IProject)container);
-			} else {
-				return getJavaProject(container.getProject());
-			}
-		}).filter(p -> p != null).collect(Collectors.toSet());
+		IProject[] projects = workspaceRoot.getProjects();
+		IPath parent = filePathFromURI(rootFolderURI);
+		return Arrays.stream(projects)
+				.filter(p -> parent.isPrefixOf(p.getLocation()))
+				.map(p -> getJavaProject(p))
+				.filter(p -> p != null)
+				.collect(Collectors.toSet());
 	}
 	
 	public static boolean isJavaProject(IProject project) {
@@ -56,6 +58,13 @@ public class ProjectUtils {
             return JavaCore.create(project);
         }
         return null;
+    }
+    
+    public static IPath filePathFromURI(URI uri) {
+    	if ("file".equals(uri.getScheme())) {
+			return Path.fromOSString(Paths.get(uri).toString());
+		}
+		return null;
     }
 
 }

